Dockerfile

```diff
@@ -1,36 +1,186 @@
-FROM quay.io/refgenomics/docker-scipy-notebook:v0.1.6
-MAINTAINER Nick <nick@onecodex.com>
-
-# Add jessie-backports repository, install bwa 0.7.13 and samtools 1.3.1
-USER 0
-RUN echo "deb http://ftp.debian.org/debian jessie-backports main" > /etc/apt/sources.list.d/bioinformatics.list \
-    && apt-get -y update \
-    && apt-get -t jessie-backports install -y "samtools" \
-    && apt-get -t jessie-backports install -y "bwa" \
-    && apt-get -t jessie-backports install -y libffi-dev libcairo2-dev libpango1.0-dev  \
-    && apt-get clean
+# Copyright (c) Reference Genomics, Inc.
+# Distributed under the terms of the Modified BSD License.
+# Extended from github.com/jupyter/docker-stacks
+# See also http://blog.dscpl.com.au/2016/01/roundup-of-docker-issues-when-hosting.html
 
-# Add assets
-RUN mkdir /opt/onecodex/
-ADD install/* /opt/onecodex/
-RUN chmod +x /opt/onecodex/notebook_report.py && \
-    ln -s /opt/onecodex/notebook_report.py /usr/local/bin/notebook_report.py
+FROM python:3.6.6-slim-stretch
+
+MAINTAINER Nick Greenfield <nick@onecodex.com>
+
+USER root
+
+# Install all OS dependencies for notebook server that starts but lacks all
+# features (e.g., download as all possible file formats)
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && apt-get install -yq --no-install-recommends \
+    wget \
+    bzip2 \
+    ca-certificates \
+    gnupg \
+    apt-transport-https \
+    sudo \
+    locales \
+    git \
+    vim \
+    build-essential \
+    python-dev \
+    unzip \
+    fonts-dejavu \
+    gfortran \
+    gcc \
+    cmake \
+    curl \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-USER 1000
+RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
+    locale-gen
+
+# Install Tini
+RUN wget --quiet https://github.com/krallin/tini/releases/download/v0.9.0/tini && \
+    echo "faafbfb5b079303691a939a747d7f60591f2143164093727e870b289a44d9872 *tini" | sha256sum -c - && \
+    mv tini /usr/local/bin/tini && \
+    chmod +x /usr/local/bin/tini
+
+# Configure environment
+ENV SHELL /bin/bash
+ENV NB_USER jovyan
+ENV NB_UID 1000
+ENV HOME /home/$NB_USER
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+ENV PATH $HOME/.local/bin:$PATH
+
+# Create jovyan user with UID=1000 and in the root group
+# See https://github.com/jupyter/docker-stacks/issues/188
+RUN useradd -m -s /bin/bash -N -u $NB_UID -g 0 $NB_USER
+
+# Setup jovyan home directory
+RUN mkdir /home/$NB_USER/work && \
+    mkdir /home/$NB_USER/.jupyter && \
+    mkdir -p -m 770 /home/$NB_USER/.local/share/jupyter && \
+    echo "cacert=/etc/ssl/certs/ca-certificates.crt" > /home/$NB_USER/.curlrc
 
 # Update pip
 RUN pip install --upgrade pip
 
-# Jupyter notebook extensions
-RUN pip install jupyter_contrib_nbextensions && \
-    jupyter contrib nbextension install --user && \
+# Numpy must be installed first
+RUN pip install numpy==1.15.4
+
+# Install Python 3 packages
+RUN pip install \
```

Comment:
These are a bunch of very specific, pinned versions that aren't the latest. Let's update anything that we actually care about and try to cut this list down a bit? 

Specific things that'd be good to ship and are _not_ themselves in `onecodex` (or will be removed):
- `matplotlib`
- `seaborn`
- `numba` and `numexpr`
- `cython`
- `bokeh`?
- `biopython` (ugh)
- `scikit-learn` (or maybe we do have that in requirements?)
- `xlrd`
- `weasyprint`

I'd be cool to ditch at least: `beautifulsoup4`, `vincent`, and `sqlalchemy` unless I'm missing something. Easy to add back in down the line.

Reply:
I'm also good removing `skimage`.

Reply:
Previously we'd basically mashed up [this Scipy notebook](https://github.com/jupyter/docker-stacks/tree/master/scipy-notebook) and [this R one](https://github.com/jupyter/docker-stacks/tree/master/r-notebook). [This](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile) is the original base notebook.

Reply:
Okey dokey, trimmed the packages down and pinned more recent versions. Users can always do something like `!pip install --user package` if they need something, I guess.